### ISSUE TITLE
batch size limit in GUI

### ISSUE THREAD

**Issue by WeissShahaf** [talmolab/sleap#1511]
# Bug description
batch size limited to 2 digits in GUI.
don't find a way to set in CLI 

## Expected behaviour
I have a GPU with 48 GB, and 720p movies, where the animal fits perfectly with no scaling. and can run at batchsize 90. using 33GB of VRAM. wanted to scale down and increase batchsize. 

## Your personal set up
WIN10, GPU: A6000

 mamba install 1.3.3

- OS: WIN10

- Version(s): 1.3.3 python 3.7

- SLEAP installation method (listed [here](https://sleap.ai/installation.html#)): used mamba  install lines from site
  - [X ] [Conda from package](https://sleap.ai/installation.html#conda-package)
  - [ ] [Conda from source](https://sleap.ai/installation.html#conda-from-source)
  - [ ] [pip package](https://sleap.ai/installation.html#pip-package)
  - [ ] [Apple Silicon Macs](https://sleap.ai/installation.html#apple-silicon-macs)

<details><summary>Environment packages</summary>

```
# paste output of `pip freeze` or `conda list` here
```

</details>

<details><summary>Logs</summary>

```
# paste relevant logs here, if any
```

</details>

## How to reproduce
try to input batch size over 99 in GUI

**Comment by WeissShahaf** [comment]
Great, thank you! @roomrys 

Now training at batchsize 512 with no issue. after scaling at 0.3, and unet with only 8 downsampling filters to keep the receptive field around the animal. Loss is at 0.000064 after 12196 batches. 

question: Is there a downside to reducing the number downsampling steps like that? besides losing information due to scaling


Thanks,
Shahaf

**Comment by roomrys** [comment]
Hi @WeissShahaf,

There are a few downsides if you decrease the number of down-sampling blocks by too much...

---
### Context
Deeper networks can capture more contextual information, which is helps with key points detection. Reducing down-sampling steps can limit the context the model can capture and make it more difficult to detect body parts.

### ... detour to Model Complexity (more to do with width of network)
On a similar note, since the number of features increases with the number of layers used, the representational capacity of the model will be decreased. Unless this is accounted for in the number of "Filters" (or kernels) used for each layer, fine-grained features will go unnoticed.
> In the SLEAP GUI, we use "Filters" to refer to the number of kernels used in each layer to extract features. We use "Max Stride" to adjust the number of down-sampling blocks used. Similarly "Output Stride" adjusts the number of up-sampling blocks used.

### Scalability/depth of detection
If the size of your animal in the field of view varies significantly in scale, reducing down-sampling steps might make the model less robust to such variations. You may need to augment your data during preprocessing to handle scale differences, but this would depend on your data.

### Saves on memory/computation
One advantage of reducing down-sampling steps is that it can reduce memory and computational requirements which helps with large batch sizes like 512.

---
...but I would encourage your experimentation to find those limits which would vary on a dataset-to-dataset basis!

Thanks,
Liezl

**Comment by roomrys** [comment]
Long, awaited, but finally integrated. This has been included in the SLEAP 1.4.1 release!
